components/brave_wallet/browser/eth_allowance_manager.cc

```diff
@@ -0,0 +1,409 @@
+#include "components/prefs/scoped_user_pref_update.h"
+
+namespace {
+constexpr int kTopicElementsCountToCheck = 3;
+constexpr char kLastBlockNumber[] = "last_block_number";
+constexpr char kAllowanceFound[] = "allowances_found";
+constexpr char kApproverAddress[] = "approver_address";
+constexpr char kContractAddress[] = "contract_address";
+constexpr char kSpenderAddress[] = "spender_address";
+constexpr char kAmount[] = "amount";
+
+constexpr char kAddress[] = "address";
+constexpr char kTopics[] = "topics";
+constexpr char kFromBlock[] = "fromBlock";
+constexpr char kToBlock[] = "toBlock";
+constexpr char kEarliestBlock[] = "earliest";
+constexpr char kLatestBlock[] = "latest";
+
+constexpr char kApprovalTopicHash[] = "Approval(address,address,uint256)";
+
+// Returns vector of chain_ids supported by allowance discovering.
+const std::vector<std::string>& GetChainIdsForAlowanceDiscovering() {
+  static base::NoDestructor<std::vector<std::string>> supported_chain_ids(
+      {brave_wallet::mojom::kMainnetChainId,
+       brave_wallet::mojom::kPolygonMainnetChainId,
+       brave_wallet::mojom::kAvalancheMainnetChainId,
+       brave_wallet::mojom::kCeloMainnetChainId,
+       brave_wallet::mojom::kArbitrumMainnetChainId,
+       brave_wallet::mojom::kOptimismMainnetChainId});
+  return *supported_chain_ids;
+}
+static std::string GetAllowanceMapKey(const std::string& contract_address,
+                                      const std::string& approver_addr,
+                                      const std::string& spender_address) {
+  return base::JoinString({contract_address, approver_addr, spender_address},
+                          "_");
+}
+
+}  // namespace
+
+namespace brave_wallet {
+EthAllowanceManager::EthAllowanceTask::EthAllowanceTask(const int& taskid)
+    : task_id(taskid) {}
+EthAllowanceManager::EthAllowanceTask::~EthAllowanceTask() = default;
+void EthAllowanceManager::EthAllowanceTask::SetResults(
+    const std::string& chain_id,
+    const uint256_t& max_block_number,
+    std::vector<mojom::AllowanceInfoPtr> alwns) {
+  allowances.insert_or_assign(
+      chain_id, std::make_tuple(max_block_number, std::move(alwns)));
+  MarkComplete();
+}
+void EthAllowanceManager::EthAllowanceTask::MarkComplete() {
+  is_completed = true;
+}
+
+EthAllowanceManager::EthAllowanceManager(BraveWalletService* wallet_service,
+                                         JsonRpcService* json_rpc_service,
+                                         KeyringService* keyring_service,
+                                         PrefService* prefs)
+    : wallet_service_(wallet_service),
+      json_rpc_service_(json_rpc_service),
+      keyring_service_(keyring_service),
+      prefs_(prefs) {}
+
+EthAllowanceManager::~EthAllowanceManager() = default;
+
+/**
+ * @brief Represents start of the allowance discovering operation
+ *
+ * @param callback callback to front end, to pass allowances list, error code
+ * and description
+ */
```

Comment:
We try to follow chromium's style for formatting and content of methods' comments. For example https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_contents.h

For this method
```
// Start of the allowance discovering operation.
```
Should be good enough comment